WcaOnRails/spec/requests/wdc_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "wdc" do
+  describe "GET /wdc" do
+    context "when not signed in" do
+      it "does not show banned competitor list" do
+        get wdc_path
+        expect(response).to be_successful
+
+        expect(response.body).not_to include "Banned"
+        expect(response.body).to include "Posts"
+      end
+    end
+
+    context "when signed in as regular user" do
+      sign_in { FactoryBot.create :user }
+
+      it 'can see banned competitors list' do
+        get wdc_path
+        expect(response).to be_successful
+
+        expect(response.body).not_to include "Banned"
+        expect(response.body).to include "Posts"
+      end
+    end
+
+    context "when signed in as a delegate" do
+      sign_in { FactoryBot.create :delegate }
+
+      it 'can see banned competitors list' do
+        get wdc_path
+        expect(response).to be_successful
+
+        expect(response.body).to include "Banned"
+        expect(response.body).to include "Posts"
+      end
+    end
+
+    context "when signed in as wdc member" do
+      sign_in { FactoryBot.create :user, :wdc_member }
+
+      it 'can see banned competitors list' do
+        get wdc_path
+        expect(response).to be_successful
+
+        expect(response.body).to include "Banned"
+        expect(response.body).to include "Posts"
+      end
+    end
```

Comment:
You could extract the `it` to some function/whatever as it's the same for the above 4 `context` blocks, but whatever you prefer =)

Reply:
Done! I pulled this out into a `assert_can_see` method that each of the tests call.